app/scripts/controllers/buildConfig.js

```diff
@@ -42,8 +42,15 @@ angular.module('openshiftConsole')
       $scope.selectedTab[$routeParams.tab] = true;
     }
 
+    // copy buildConfig and ensure it has env so that we can edit env vars using key-value-editor
+    var copyBuildConfigAndEnsureEnv = function(buildConfig) {
+      $scope.updatedBuildConfig = angular.copy(buildConfig);
+      $scope.envVars = $filter('buildStrategy')($scope.updatedBuildConfig).env || [];
```

Comment:
If you move the func after the `buildStrategy` declaration just below, you won't have the call `$filter`.

```js
$scope.envVars = buildStrategy($scope.updatedBuildConfig).env || [];
```